src/lib.rs

```diff
@@ -47,31 +72,66 @@ impl Config {
             ("state", &state),
             ("scope", &scopes),
         ];
+
         if self.redirect_url.len() > 0 {
             pairs.push(("redirect_uri", &self.redirect_url));
         }
+
         let mut url = self.auth_url.clone();
         url.query_pairs_mut().clear().extend_pairs(
             pairs.iter().map(|&(k, v)| { (k, &v[..]) })
         );
-        return url;
+
+        url
+    }
+
+    #[deprecated(since="0.4.0", note="please use `exchange_code` instead")]
```

Comment:
not sure about what version you'll release or whether I should mark it as deprecated or remove it altogether. thoughts?